File: examples/Delphi/ExCalU.pas
{$J+} {Writable constants}

unit ExCalU;

interface

uses
  Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  OvcBase, OvcCal;

type
  TForm4 = class(TForm)
    OvcCalendar1: TOvcCalendar;
    procedure OvcCalendar1GetDateEnabled(Sender: TObject; ADate: TDateTime;
      var Enabled: Boolean);
    procedure OvcCalendar1GetHighlight(Sender: TObject; ADate: TDateTime;
      var Color: TColor);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form4: TForm4;

implementation

{$R *.DFM}

procedure TForm4.OvcCalendar1GetDateEnabled(Sender: TObject;
  ADate: TDateTime; var Enabled: Boolean);
begin
  Enabled := (DayOfWeek(ADate) <> 3);
end;

procedure TForm4.OvcCalendar1GetHighlight(Sender: TObject;
  ADate: TDateTime; var Color: TColor);
begin
  if DayOfWeek(ADate) = 3 then
    Color := clGreen;
end;

end.

File: examples/CBuildr3/ExCalu.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExCalu.h"
//---------------------------------------------------------------------------
#pragma link "OvcCal"
#pragma link "OvcBase"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::OvcCalendar1GetDateEnabled(TObject *Sender,
	TDateTime ADate, bool &Enabled)
{
  Enabled = (DayOfWeek(ADate) != 3);
}
//---------------------------------------------------------------------------
void __fastcall TForm1::OvcCalendar1GetHighlight(TObject *Sender,
	TDateTime ADate, TColor &Color)
{
  if (DayOfWeek(ADate) == 3)  {
    Color = clGreen;
  	}
}
//--------------------------------------------------------------------------- 
